### DISCUSSION TITLE
popup-page reload, an exception occurs.

### DISCUSSION THREAD

**Discussion by JaehwanMango** [CommunityToolkit/Maui#700] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/107899309/196847558-7fd687a0-0625-4417-81ad-def2fd8b5b9e.png)
If popup-page close and reload, an exception occurs.

```
var popup = new SimplePopup();

this.ShowPopup(popup);
```
[Reference Docs](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/views/popup)

I don't  like `new page()`
(Cuz instance Page in ViewModel...)
So use Ioc
```
builder.Services.AddTransient<Popup>();

// ViewModel
 ...
public MyViewMolde(Popup popup)
{ _popup = popup; }

Private async Task ItemClick()
{
    await Application.Current.MainPage.ShowPopupAsync(_popup ).ConfigureAwait(false);
}
```

1. open the Popup page
2. Close popup page
3. open again popup page
4. Exception disposed...

why popup-page disposed? Cuz closed?
need be instance when open Popup-page?

Can i block the disposed?

Tooo hard MVVM....

**Comment by pictos** [comment, marked as answer]
@JaehwanMango this intended by design, when the popup disappears, it get disposed is ready to be collected. So instead of reusing the Popup you should create a new one.

**Reply by JaehwanMango** [reply]
> @JaehwanMango this intended by design, when the popup disappears, it get disposed is ready to be collected. So instead of reusing the Popup you should create a new one.

@pictos  Thanks
reply so fast!!! awesome

**Reply by pictos** [reply]
You're welcome... But let me know, there's a scenario that this is bad for you? Or it's fine to create a new instance everytime?

**Reply by JaehwanMango** [reply]
@pictos Thanks for service

This scenario is bad for me
I can't control the life cycle => this mean **no options**

It's not worth if popup has to dispose

**Comment by CliffAgius** [comment]
Closed as answered.
